app/nexus/internal/initialization/recovery/recovery.go

```diff
@@ -72,34 +71,28 @@ func RecoverBackingStoreUsingKeeperShards(source *workloadapi.X509Source) {
 	_, err := retry.Do(ctx, func() (bool, error) {
 		log.Log().Info(fName, "message", "retry:"+time.Now().String())
 
-		recoverySuccessful := iterateKeepersAndTryRecovery(
+		initSuccessful := iterateKeepersAndInitializeState(
 			source, successfulKeeperShards,
 		)
-		if recoverySuccessful {
-			log.Log().Info(fName, "message", "Recovery successful")
+		if initSuccessful {
+			log.Log().Info(fName, "message", "Initialization successful.")
 			return true, nil
 		}
 
-		log.Log().Warn(fName, "message", "Recovery unsuccessful. Will retry.")
-		log.Log().Warn(fName, "message",
-			fmt.Sprintf(
-				"Successful keepers: %d", len(successfulKeeperShards),
-			),
-		)
 		log.Log().Warn(fName,
-			"message", "!!! YOU MAY NEED TO MANUALLY BOOSTRAP !!!")
-		log.Log().Info(fName,
-			"message", "Waiting for keepers to respond")
+			"message", "Initialization unsuccessful. Will retry.",
+			"keepersSoFar", len(successfulKeeperShards),
+		)
 		return false, ErrRecoveryRetry
 	},
 		retry.WithBackOffOptions(
 			retry.WithMaxInterval(env.RecoveryOperationMaxInterval()),
-			retry.WithMaxElapsedTime(env.RecoveryOperationTimeout()),
+			retry.WithMaxElapsedTime(0), // Retry forever.
```

Comment:
maybe we can add a configurable "bootsrap time out" as an env var later. For now, the bootsrap app will try to bootstrap the thing in an exponentially-backing-off loop until it succeeds.